### PR TITLE
Move gethrtime() calls out of vdev queue lock.

### PR DESCRIPTION
**Pull request by amotin**
This dramatically reduces the lock contention on systems with slower
(non-TSC) timecounters.  With TSC the difference is minimal, but since
this lock is pretty congested, any improvement counts.  Plus I don't
see any reason to do it under the lock other than the latency of the
lock itself, which this change actually reduces.

### How Has This Been Tested?
On 40-thread FreeBSD system switched to use ACPI timecounter instead of TSC on heavy uncached 4KB ZVOLs reads profiler shows reduction of CPU time spent spinning on the vdev queue lock from 25% to almost zero.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/zfs/vdev_queue.c [hunk 1]
```diff
@@ -936,14 +936,13 @@ vdev_queue_io_done(zio_t *zio)
 	vdev_queue_t *vq = &zio->io_vd->vdev_queue;
 	zio_t *nio;
 
-	mutex_enter(&vq->vq_lock);
+	hrtime_t now = gethrtime();
+	vq->vq_io_complete_ts = now;
+	vq->vq_io_delta_ts = zio->io_delta = now - zio->io_timestamp;
```
Comment (ahrens) [inline module/zfs/vdev_queue.c:941]: I think that these assignments into `vq` should still take place under the `vq_lock`

**Review by ahrens — APPROVED** [review]
This seems fine.  Out of curiosity, what are the most common systems that don't have a fast gethrtime() by default?  Or are people disabling TSC for some reason?

File: module/zfs/vdev_queue.c (hunk 1 above)
Comment (amotin) [reply module/zfs/vdev_queue.c:941]: As I see, these variables are read without holding the vq_lock, so if they wish to get torn on 32-bit system, they will, and vq_lock won't help.

**Comment by amotin** [comment]
There were plenty of systems with problematic TSC over the years: changing rate, stopping in deep idle, not coherent on SMP, slow in VMs, etc.  Last time it is better, but why depend on something if it can be avoided?

**Review by behlendorf — APPROVED** [review]

**Pull request merged**